text/profiles/0237-profiles-data-model.md

```diff
@@ -0,0 +1,1015 @@
+
+##### Field `attribute_set_indices`
+
+List of indices referring to AttributeSets in the Profile's attribute set table. Each attribute set corresponds to a Stacktrace in stacktrace_indices list. Length must match stacktrace_indices length. [Optional]
+
+##### Field `values`
+
+List of values. Each value corresponds to a Stacktrace in stacktrace_indices list. Length must match stacktrace_indices length.
+
+##### Field `timestamps`
+
+List of timestamps. Each timestamp corresponds to a Stacktrace in stacktrace_indices list. Length must match stacktrace_indices length.
+</details>
+
+#### `Sample` structure
+
+Sample is an ephemeral structure. It is not explicitly represented as a protobuf message, instead it is represented by stacktraces, links, attribute sets, values and timestamps tables in `ProfileType` message. The connection is based on the order of the elements in the corresponding tables. For example, AttributeSet with index 1 corresponds to a Stacktrace located at index 1 in stacktraces table, and a Value located at index 1 in values table. Together they form a Sample.
+
+#### Message `Stacktrace`
+
+A stacktrace is a sequence of locations. Order of locations goes from callers to callees. Many stacktraces will point to the same locations. The link between stacktraces, attribute sets, links, values and timestamps is implicit and is based on the order of the elements in the corresponding tables in ProfileType message.
+
+#### Message `AttributeSet`
+
+AttributeSet represents a set of attributes. Multiple Samples, Locations and Mappings may have the same attributes and that's why this is a separate message. These are stored in a lookup table in a Profile. These are referenced by index from other messages.
+
+<details>
+<summary>Field Descriptions</summary>
+
+##### Field `attributes`
+
+Attributes associated with a specific Sample, Location or a Mapping.
+attributes is a collection of key/value pairs. Note, global attributes
+like server name can be set using the resource API.
+
+##### Field `dropped_attributes_count`
+
+dropped_attributes_count is the number of attributes that were discarded. Attributes
+can be discarded because their keys are too long or because there are too many
+attributes. If this value is 0, then no attributes were dropped.
+</details>
+
+#### Message `Link`
+
+Represents a relationship between a Sample (ephemeral structure represented by references to a Stacktrace, AttributeSet, Link + value and a timestamp) and a trace span. This allows for linking between specific Samples within a profile and traces.
+
+<details>
+<summary>Field Descriptions</summary>
+
+##### Field `trace_id`
+
+A unique identifier of a trace that this linked span is part of. The ID is a
+16-byte array.
+
+##### Field `span_id`
+
+A unique identifier for the linked span. The ID is an 8-byte array.
```

Comment:
You may want to call out that trace_id/span_id are optional. Also in log data model we allow trace_id to present when span_id is unspecified (linking to the trace but not to the specific span). We should probably make that legal for profiles too.